compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -799,6 +799,27 @@ fn link_natively(
             continue;
         }
 
+        // Check if linking failed with an error message that indicates the driver didn't recognize
+        // the `-fuse-ld=lld` option. If so, re-perform the link step without it. This avoids having
+        // to spawn multiple instances on the happy path to do version checking, and ensures things
+        // keep working on the tier 1 baseline of GLIBC 2.17+. That is generally understood as GCCs
+        // circa RHEL/CentOS 7, 4.5 or so, whereas lld support was added in GCC 9.
+        if matches!(flavor, LinkerFlavor::Gnu(Cc::Yes, Lld::Yes))
+            && unknown_arg_regex.is_match(&out)
+            && out.contains("-fuse-ld=lld")
+            && cmd.get_args().iter().any(|e| e.to_string_lossy() == "-fuse-ld=lld")
```

Comment:
```suggestion
            && cmd.get_args().iter().any(|e| e.to_str() == Some("-fuse-ld=lld"))
```

Reply:
This code is taken from the same procedure above and below. Care to elaborate?

Reply:
It does a tiny bit less work in a corner case. But if the code above looks the same, I guess you don't need to adjust it. I just didn't see any use case for a lossy conversion.

Reply:
Yeah the utf8 conversion or allocation costs are not going to be super important in this error path. Let's keep things consistent for now in this PR where I'm trying to fix a p-high issue. But IMO, if you have time, feel free to open a PR changing this existing codepath, it will not hurt to discuss this with other people. 

I'll resolve this for now.